### ISSUE TITLE
goose keeps remembering old file content

### ISSUE THREAD

**Issue by michaelneale** [block/goose#84]
Even though it knows file has changed, it is unable to "forget" the old content of the file from its context. 

Result: often old unwanted changes make their way back into a file, if you have edited it outside of the goose session. 

Solution: probably need to prune past content of the file in the current session once it is detected to have changed, if at all possible.

**Comment by michaelneale** [comment]
I think this may be able to be addressed with a moderator: a moderator which drops old content smartly especially if it knows there is a new version of the file.

**Comment by baxen** [comment]
Agree! pruning reads from the history based on path probably improves performance and should also drop context naturally quite a bit
